File: illus_the_relativistic_eccentricity_correction.py
# Integrated Navigation Systems — Chapter 6: Satellite Navigation Fundamentals (GNSS)
# Numerical Illustration — The relativistic eccentricity correction
# Companion code (educational reference implementation; see the book for context).

import numpy as np
F=-4.442807633e-10; dtr=F*0.01*np.sqrt(26560e3)
print(f'max {abs(dtr)*1e9:.1f} ns = {abs(dtr)*299792458:.2f} m')  # 22.9 6.86

File: illus_the_relativistic_eccentricity_correction.m
% Integrated Navigation Systems — Chapter 6: Satellite Navigation Fundamentals (GNSS)
% Numerical Illustration — The relativistic eccentricity correction
% Companion code (educational reference implementation; see the book for context).

F=-4.442807633e-10; e=0.01; a=26560e3;
dtr=F*e*sqrt(a); fprintf('max %.1f ns = %.2f m\n',abs(dtr)*1e9,abs(dtr)*299792458); % 22.9, 6.86
